src/lasp_core.erl

```diff
@@ -182,20 +182,23 @@ read(Id, Store) ->
 %%      bound.
 %%
 %%      This operation blocks until `Threshold' has been reached.
+%%      As read/6 runs inside a lasp process, we have to block
+%%      outside, or else it may get killed when used in combination
+%%      with other lasp processes (for example, when used as the read
+%%      function of another lasp process).
 %%
 -spec read(id(), value(), store()) -> {ok, var()}.
 read(Id, Threshold, Store) ->
-    Self = self(),
-    ReplyFun = fun({Id1, Type, Metadata, Value}) ->
-                       {ok, {Id1, Type, Metadata, Value}}
-               end,
-    BlockingFun = fun() ->
-                receive
-                    X ->
-                        X
-                end
-            end,
-    read(Id, Threshold, Store, Self, ReplyFun, BlockingFun).
+    ReplyFun = fun
+        ({error, _}) -> error;
```

Comment:
Why not return `{error, Reason}`?  You've got a reason here, and it's probably good to propagate it forward, at least for logging.